Spanner: 'test_transaction_batch_update_and_execute_dml' systest flakes with 'None' status code.

Similar to #7504, but this time the status code is `None`.

From [this CI failure](https://source.cloud.google.com/results/invocations/ff8f8339-1a05-41e6-ac3a-6ed36f0061e1/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fspanner/log):

```python
_________ TestSessionAPI.test_transaction_batch_update_and_execute_dml _________

self = <tests.system.test_system.TestSessionAPI testMethod=test_transaction_batch_update_and_execute_dml>

    def test_transaction_batch_update_and_execute_dml(self):
        retry = RetryInstanceState(_has_all_ddl)
        retry(self._db.reload)()

        session = self._db.session()
        session.create()
        self.to_delete.append(session)

        with session.batch() as batch:
            batch.delete(self.TABLE, self.ALL)

        insert_statements = list(self._generate_insert_statements())
        update_statements = [
            (
                "UPDATE contacts SET email = @email " "WHERE contact_id = @contact_id;",
                {"contact_id": 1, "email": "phreddy@example.com"},
                {"contact_id": Type(code=INT64), "email": Type(code=STRING)},
            )
        ]

        delete_statement = "DELETE contacts WHERE TRUE;"

        def unit_of_work(transaction, self):
            rows = list(transaction.read(self.TABLE, self.COLUMNS, self.ALL))
            self.assertEqual(rows, [])

            status, row_counts = transaction.batch_update(
                insert_statements + update_statements
            )
            self._check_batch_status(status.code)
            self.assertEqual(len(row_counts), len(insert_statements) + 1)
            for row_count in row_counts:
                self.assertEqual(row_count, 1)

            row_count = transaction.execute_update(delete_statement)

            self.assertEqual(row_count, len(insert_statements))

>       session.run_in_transaction(unit_of_work, self)

tests/system/test_system.py:874:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
google/cloud/spanner_v1/session.py:299: in run_in_transaction
    return_value = func(txn, *args, **kw)
tests/system/test_system.py:865: in unit_of_work
    self._check_batch_status(status.code)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

status_code = 10

    @staticmethod
    def _check_batch_status(status_code):
        if status_code != code_pb2.OK:
>           raise exceptions.from_grpc_status(status_code, "batch_update failed")
E           google.api_core.exceptions.GoogleAPICallError: None batch_update failed
```

--- a/spanner/tests/system/test_system.py
+++ b/spanner/tests/system/test_system.py
@@ -790,8 +790,8 @@ class TestSessionAPI(unittest.TestCase, _TestData):
         # [END spanner_test_dml_with_mutation]
 
     @staticmethod
-    def _check_batch_status(status_code):
-        if status_code != code_pb2.OK:
+    def _check_batch_status(status_code, expected=code_pb2.OK):
+        if status_code != expected:
             grpc_status_code = _STATUS_CODE_TO_GRPC_STATUS_CODE[status_code]
             call = FauxCall(status_code)
             raise exceptions.from_grpc_status(
@@ -915,7 +915,7 @@ class TestSessionAPI(unittest.TestCase, _TestData):
             status, row_counts = transaction.batch_update(
                 [insert_statement, update_statement, delete_statement]
             )
-            self.assertEqual(status.code, code_pb2.INVALID_ARGUMENT)
+            self._check_batch_status(status.code, code_pb2.INVALID_ARGUMENT)
             self.assertEqual(len(row_counts), 1)
             self.assertEqual(row_counts[0], 1)
 

--- a/spanner/tests/system/test_system.py
+++ b/spanner/tests/system/test_system.py
@@ -794,7 +794,7 @@ class TestSessionAPI(unittest.TestCase, _TestData):
         if status_code != code_pb2.OK:
             grpc_status_code = _STATUS_CODE_TO_GRPC_STATUS_CODE[status_code]
             call = FauxCall(status_code)
-            exc = exceptions.from_grpc_status(
+            raise exceptions.from_grpc_status(
                 grpc_status_code, "batch_update failed", errors=[call]
             )
 

--- a/spanner/tests/system/test_system.py
+++ b/spanner/tests/system/test_system.py
@@ -24,6 +24,7 @@ import unittest
 import uuid
 
 import pytest
+import grpc
 from google.rpc import code_pb2
 
 from google.api_core import exceptions
@@ -65,6 +66,10 @@ else:
 EXISTING_INSTANCES = []
 COUNTERS_TABLE = "counters"
 COUNTERS_COLUMNS = ("name", "value")
+
+_STATUS_CODE_TO_GRPC_STATUS_CODE = {
+    member.value[0]: member for member in grpc.StatusCode
+}
 
 
 class Config(object):
@@ -787,7 +792,10 @@ class TestSessionAPI(unittest.TestCase, _TestData):
     @staticmethod
     def _check_batch_status(status_code):
         if status_code != code_pb2.OK:
-            raise exceptions.from_grpc_status(status_code, "batch_update failed")
+            grpc_status_code = _STATUS_CODE_TO_GRPC_STATUS_CODE[status_code]
+            call = FauxCall(status_code)
+            exc  = exceptions.from_grpc_status(
+                grpc_status_code, "batch_update failed", errors=[call])
 
     def test_transaction_batch_update_success(self):
         # [START spanner_test_dml_with_mutation]
@@ -2190,3 +2198,22 @@ class _ReadAbortTrigger(object):
     def handle_abort(self, database):
         database.run_in_transaction(self._handle_abort_unit_of_work)
         self.handler_done.set()
+
+
+class FauxCall(object):
+
+    def __init__(self, code, details="FauxCall"):
+        self._code = code
+        self._details = details
+
+    def initial_metadata(self):
+        return {}
+
+    def trailing_metadata(self):
+        return {}
+
+    def code(self):
+        return self._code
+
+    def details(self):
+        return self._details

--- a/spanner/tests/system/test_system.py
+++ b/spanner/tests/system/test_system.py
@@ -794,8 +794,9 @@ class TestSessionAPI(unittest.TestCase, _TestData):
         if status_code != code_pb2.OK:
             grpc_status_code = _STATUS_CODE_TO_GRPC_STATUS_CODE[status_code]
             call = FauxCall(status_code)
-            exc  = exceptions.from_grpc_status(
-                grpc_status_code, "batch_update failed", errors=[call])
+            exc = exceptions.from_grpc_status(
+                grpc_status_code, "batch_update failed", errors=[call]
+            )
 
     def test_transaction_batch_update_success(self):
         # [START spanner_test_dml_with_mutation]
@@ -2201,7 +2202,6 @@ class _ReadAbortTrigger(object):
 
 
 class FauxCall(object):
-
     def __init__(self, code, details="FauxCall"):
         self._code = code
         self._details = details

